Sources/SpeziStudy/Study Manager/StudyManager.swift

```diff
@@ -556,6 +530,59 @@ extension StudyManager {
 }
 
 
+extension StudyManager {
+    @MainActor
+    private func setupStudyBackgroundComponents(for enrollments: some Collection<StudyEnrollment>) async throws {
+        try await startBackgroundHealthDataCollection(for: enrollments)
+    }
+    
+    @MainActor
+    private func tearDownStudyBackgroundComponents(for enrollments: some Collection<StudyEnrollment>) async {
+        await stopBackgroundHealthDataCollection(for: enrollments)
+    }
+    
+    
+    private func allCollectedSampleTypes(in enrollments: some Collection<StudyEnrollment>) -> [any AnySampleType] {
+        enrollments.reduce(into: []) { sampleTypes, enrollment in
+            guard let healthCollectionComponents = enrollment.studyBundle?.studyDefinition.healthDataCollectionComponents else {
+                return
+            }
+            for component in healthCollectionComponents {
+                sampleTypes.append(contentsOf: component.sampleTypes)
+            }
+        }
+    }
+    
+    @MainActor
+    private func startBackgroundHealthDataCollection(for enrollments: some Collection<StudyEnrollment>) async throws {
+        func setupSampleCollection<Sample>(_ sampleType: some AnySampleType<Sample>) async {
+            let sampleType = SampleType(sampleType)
+            await healthKit.addHealthDataCollector(CollectSample(
+                sampleType,
+                start: .automatic,
+                continueInBackground: true
+            ))
+        }
+        for sampleType in allCollectedSampleTypes(in: enrollments) {
+            await setupSampleCollection(sampleType)
+        }
+        // we want to request HealthKit auth once, at the end, for everything we just registered.
+        try await healthKit.askForAuthorization()
+    }
+    
+    @MainActor
+    private func stopBackgroundHealthDataCollection(for enrollments: some Collection<StudyEnrollment>) async {
+        func setupSampleCollection<Sample>(_ sampleType: some AnySampleType<Sample>) async {
+            let sampleType = SampleType(sampleType)
+            await healthKit.resetSampleCollection(for: sampleType)
+        }
+        for sampleType in allCollectedSampleTypes(in: enrollments) {
```

Comment:
I think this might be a bit of an issue in the long run: What if an app has also requested a sample collection by itself and wants to keep that going, it might have just overlapped with the one from a study. I don't think the that HK module would cover that. 

In general, would also indicate that we might want to change some of SpeziHK data collection structure in the long run, e.g., not funnel everything to the Standard (which might be removed in the long run anyways as we aim to move away from a singleton central class) and rather requests specific delivery methods for each time we request the collection of HealthKit samples to a type that serves as a delegate.

Both not things that we might want to solve here, but things we should note down in issues and address later on?